src/test/c_api_tests.cpp

```diff
@@ -944,7 +932,67 @@ TEST_F(CAPIInference, NegativeInference) {
     OVMS_ServerDelete(cserver);
     OVMS_ServerDelete(nullptr);
 }
-
+TEST_F(CAPIInference, String) {
+    using std::string;
+    ServerGuard serverGuard("/ovms/src/test/c_api/config_string.json");
+    OVMS_InferenceRequest* request{nullptr};
+    const string MODEL_NAME = "passthrough";
+    const string INPUT_NAME = "my_name";
+    const string OUTPUT_NAME = "my_name";
+    ASSERT_CAPI_STATUS_NULL(OVMS_InferenceRequestNew(&request, serverGuard.server, MODEL_NAME.c_str(), 1));
+    ASSERT_NE(nullptr, request);
+    std::vector<string> data{{"Intel"}, {"CCG"}};
+    std::vector<int64_t> inShape{(int64_t)data.size()};
+    ASSERT_CAPI_STATUS_NULL(OVMS_InferenceRequestAddInput(request, INPUT_NAME.c_str(), OVMS_DATATYPE_STRING, inShape.data(), inShape.size()));
+    // setting buffer
+    uint32_t notUsedNum = 0;
+    ASSERT_CAPI_STATUS_NULL(OVMS_InferenceRequestInputSetData(request, INPUT_NAME.c_str(), reinterpret_cast<void*>(&data[0]), int64_t(data.size()), OVMS_BUFFERTYPE_CPU, notUsedNum));
+    //////////////////
+    //  INFERENCE
+    //////////////////
+    OVMS_InferenceResponse* response = nullptr;
+    SPDLOG_ERROR("ER:{}", (void*)&data);
+    ASSERT_CAPI_STATUS_NULL(OVMS_Inference(serverGuard.server, request, &response));
+    // verify GetOutputCount
+    uint32_t outputCount = 42;
+    ASSERT_CAPI_STATUS_NULL(OVMS_InferenceResponseOutputCount(response, &outputCount));
+    ASSERT_EQ(outputCount, 1);
+    // verify GetOutput
+    const void* voutputData;
+    size_t bytesize = 42;
+    uint32_t outputId = 0;
+    OVMS_DataType datatype = (OVMS_DataType)199;
+    const int64_t* shape{nullptr};
+    size_t dimCount = 42;
+    OVMS_BufferType bufferType = (OVMS_BufferType)199;
+    uint32_t deviceId = 42;
+    const char* outputName{nullptr};
+    ASSERT_CAPI_STATUS_NULL(OVMS_InferenceResponseOutput(response, outputId, &outputName, &datatype, &shape, &dimCount, &voutputData, &bytesize, &bufferType, &deviceId));
+    ASSERT_EQ(string(OUTPUT_NAME.c_str()), outputName);
+    EXPECT_EQ(datatype, OVMS_DATATYPE_STRING);
+    EXPECT_EQ(dimCount, 1);
+    EXPECT_EQ(bufferType, OVMS_BUFFERTYPE_CPU);
+    EXPECT_EQ(deviceId, 0);
+    EXPECT_EQ(bytesize, sizeof(string) * data.size());
+    EXPECT_EQ(data.size(), bytesize / sizeof(string));
+    EXPECT_TRUE(std::equal(data.begin(), data.end(), reinterpret_cast<const string*>(voutputData)));
+    // now perform second inference and then check if first output is still ok
+    OVMS_InferenceResponse* response2 = nullptr;
+    const std::vector<std::string> originalData = data;
+    // request still points to original data
+    data[0] = "Habana";
+    data[1] = "Gaudi";
+    ASSERT_CAPI_STATUS_NULL(OVMS_Inference(serverGuard.server, request, &response2));
```

Comment:
I cant see the test that allows setting output buffer for string to write to. Is it supported? @atobiszei

Reply:
For now it is not. Added assert.